src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineDraft.java

```diff
@@ -0,0 +1,240 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.ui.stepsdesign;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.ui.utils.Commands;;
+import io.cdap.cdap.ui.utils.Constants;
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.pages.actions.CdfPluginPropertiesActions;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+import org.openqa.selenium.WebElement;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.UUID;
+
+public class PipelineDraft {
+
+  public static final Gson GSON = new Gson();
+  public static final String DUMMY_PROJECT = "DummyProject";
+  public static final String DUMMY_DATASET = "DummyDataset";
+  public static final String PIPELINE_TYPE = "cdap-data-pipeline";
+
+  public static String pipelineName = "";
+  public static String pipelineDraftName = "";
+  public static String newPipelineDraftName = "";
+  public static String oldApiPipelineDraftName = "";
+  public static String draftId = "";
+
+
+  @When("Create simple pipeline")
+  public void createSimplePipeline() {
+    Commands.createSimplePipeline();
+  }
+
+  @Then(" Open source node property")
+  public void openSourceNodeProperty() {
+    CdfStudioActions.navigateToPluginPropertiesPage(Commands.simpleSourceNode.getNodeName());
+  }
+
+  @Then("Add DummyProject as dataset project")
+  public void addProject() throws InterruptedException {
+    WebElement projectField = Helper.locateElementByXPath("//input[@data-testid='datasetProject']");
+    ElementHelper.replaceElementValue(projectField, DUMMY_PROJECT);
+  }
+
+  @Then("Add DummyDataset as dataset")
+  public void addDataset() {
+    WebElement datasetField = Helper.locateElementByXPath("//input[@data-testid='dataset']");
+    ElementHelper.clearElementValue(datasetField);
+    ElementHelper.replaceElementValue(datasetField, DUMMY_DATASET);
+  }
+
+  @Then("Close source node property")
+  public void closeSourceNodeProperty() {
+    CdfPluginPropertiesActions.clickCloseButton();
+  }
+
+  @Then("Fill in pipeline name")
+  public void fillInPipelineName() {
+    pipelineName = "Test_Draft_Pipeline_" + UUID.randomUUID();
+    pipelineDraftName = "pre_publish_" + pipelineName;
```

Comment:
Is it a good practice to use a global variable?

Reply:
As long as they're used within the same scenario (which is the case here), I don't think it's bad practice. Doesn't break the independency of individual tests.